components/curriculum/layout.css

```diff
@@ -1,46 +1,172 @@
-/* layout */
-.curriculum-module,
+.curriculum-content-container {
+  display: grid;
+
+  grid-template-areas:
+    "sidebar . header . toc"
+    "sidebar . body   . toc";
+  grid-template-rows: min-content auto;
+  grid-template-columns: var(--layout-2-sidebars);
+
+  justify-content: space-between;
+
+  padding-inline: var(--layout-side-padding);
+
+  .curriculum-layout__content {
+    display: contents;
+  }
+
+  .curriculum-layout__header {
+    grid-area: header;
+  }
+
+  .curriculum-layout__body {
+    grid-area: body;
+    padding-bottom: 3rem;
+  }
+
+  .curriculum-layout__sidebar {
+    grid-area: sidebar;
+    padding-top: 2rem;
+
+    @media (--screen-layout-2-sidebars) {
+      .left-sidebar {
+        padding-right: calc(var(--layout-sidebar-gap) / 2);
+        margin-right: calc(var(--layout-sidebar-gap) / -2);
+      }
+    }
+  }
+
+  .curriculum-layout__toc {
+    display: flex;
+
+    flex-wrap: wrap;
+
+    grid-area: toc;
+
+    gap: 0.5rem;
+    align-content: start;
+    align-items: start;
+    justify-content: space-between;
+  }
+
+  .curriculum-layout__sidebar .left-sidebar,
+  .curriculum-layout__toc {
+    position: sticky;
+    top: var(--sticky-header-height);
+
+    max-height: calc(100vh - var(--sticky-header-height));
+
+    overflow-y: auto;
+  }
+
+  @media (--screen-layout-1-sidebar-or-less) {
+    grid-template-areas:
+      "toc . header"
+      "toc . body";
+    grid-template-columns: var(--layout-1-sidebar-left);
+
+    .curriculum-layout__sidebar {
+      z-index: 1;
+
+      display: none;
+
+      grid-area: toc;
+
+      background: var(--color-background-page);
+    }
+  }
+
+  @media (--screen-layout-no-sidebar) {
+    display: block;
+
+    .curriculum-layout__sidebar {
```

Comment:
There's a slight gap on mobile:

<img width="750" height="1334" alt="Screen Shot 2025-07-30 at 16 09 05" src="https://github.com/user-attachments/assets/302e269d-b592-454d-8997-fcfa80bdb92d" />

Which should be fixed by removing the padding:

```suggestion
    .curriculum-layout__sidebar {
      padding-top: 0;
```